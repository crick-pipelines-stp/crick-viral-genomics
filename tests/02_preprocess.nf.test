"""
Test cases for the preprocess components of the pipeline
"""

nextflow_pipeline {
    name "Test Pipeline Components"
    script "../main.nf"

    tag "preprocess"

    // Test merging of single fastq files
    test("preprocess_cat_fastq_singles") {
        tag "preprocess/cat_fastq_singles"
        when {
            params {
                run_reporting = false

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/data/refs/covid/SARS-CoV-2.fasta"
                samplesheet = "$projectDir/tests/data/samplesheets/test_nanopore_cov_amp.csv"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            def task_count = workflow.trace.tasks().count { task ->
                task.name.contains("CAT_FASTQ")
            }
            assertAll(
                { assert workflow.success},
                { assert task_count == 2 },
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }

    // Test merging of a folder of fastq files
    test("preprocess_cat_fastq_folder") {
        tag "preprocess/cat_fastq_folder"
        when {
            params {
                run_reporting = false

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/data/refs/covid/SARS-CoV-2.fasta"
                samplesheet = "$projectDir/tests/data/samplesheets/test_nanopore_cov_amp_folder.csv"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            def task_count = workflow.trace.tasks().count { task ->
                task.name.contains("CAT_FASTQ")
            }
            assertAll(
                { assert workflow.success},
                { assert task_count == 2 },
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }

    // Test illumina read trimming and QC
    test("preprocess_trim_illumina") {
        tag "preprocess/trim_illumina"
        when {
            params {
                run_illumina_qc_trim = true
                run_reporting        = false

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/tests/data/seq_sim/influenza_A/*"
                samplesheet = "$projectDir/tests/data/samplesheets/test_illumina_flu_amp.csv"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(
                params.outdir,
                relative: true,
                includeDir: true,
                ignore: [
                    'pipeline_info/*.{html,json,txt}',
                    '**/*.fastp.log',
                ])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }

    // Test nanopore read trimming (no timming at the moment, just QC)
    test("preprocess_trim_nanopore") {
        tag "preprocess/trim_nanopore"
        when {
            params {
                run_nanopore_qc_trim = true
                run_reporting        = false

                outdir      = "$outputDir"
                viral_fasta = "$projectDir/data/refs/covid/SARS-CoV-2.fasta"
                samplesheet = "$projectDir/tests/data/samplesheets/test_nanopore_cov_amp.csv"
            }
        }
        then {
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.trace.succeeded().size(),
                    stable_name,
                    stable_path
                ).match() }
            )
        }
    }
}
