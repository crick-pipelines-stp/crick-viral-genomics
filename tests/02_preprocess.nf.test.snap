{
    "preprocess_cat_fastq_folder": {
        "content": [
            4,
            [
                "00_reference",
                "00_reference/SARS-CoV-2.fasta.fai",
                "pipeline_info"
            ],
            [
                "SARS-CoV-2.fasta.fai:md5,6806a14930eab7e2de4e9776d79bcab2"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-14T15:35:55.789287"
    },
    "preprocess_trim_illumina": {
        "content": [
            12,
            [
                "00_reference",
                "00_reference/viral_reference.merged.fasta",
                "00_reference/viral_reference.merged.fasta.fai",
                "01_qc",
                "01_qc/fastqc",
                "01_qc/fastqc/raw",
                "01_qc/fastqc/raw/2-A_SouthSudan_1029_2024_raw_1_fastqc.html",
                "01_qc/fastqc/raw/2-A_SouthSudan_1029_2024_raw_1_fastqc.zip",
                "01_qc/fastqc/raw/2-A_SouthSudan_1029_2024_raw_2_fastqc.html",
                "01_qc/fastqc/raw/2-A_SouthSudan_1029_2024_raw_2_fastqc.zip",
                "01_qc/fastqc/raw/20-A_Tajikistan_12-928_2023_raw_1_fastqc.html",
                "01_qc/fastqc/raw/20-A_Tajikistan_12-928_2023_raw_1_fastqc.zip",
                "01_qc/fastqc/raw/20-A_Tajikistan_12-928_2023_raw_2_fastqc.html",
                "01_qc/fastqc/raw/20-A_Tajikistan_12-928_2023_raw_2_fastqc.zip",
                "01_qc/fastqc/raw/51-A_Spain_1810_2024_S51_raw_1_fastqc.html",
                "01_qc/fastqc/raw/51-A_Spain_1810_2024_S51_raw_1_fastqc.zip",
                "01_qc/fastqc/raw/51-A_Spain_1810_2024_S51_raw_2_fastqc.html",
                "01_qc/fastqc/raw/51-A_Spain_1810_2024_S51_raw_2_fastqc.zip",
                "01_qc/fastqc/trimmed",
                "01_qc/fastqc/trimmed/2-A_SouthSudan_1029_2024_trimmed_1_fastqc.html",
                "01_qc/fastqc/trimmed/2-A_SouthSudan_1029_2024_trimmed_1_fastqc.zip",
                "01_qc/fastqc/trimmed/2-A_SouthSudan_1029_2024_trimmed_2_fastqc.html",
                "01_qc/fastqc/trimmed/2-A_SouthSudan_1029_2024_trimmed_2_fastqc.zip",
                "01_qc/fastqc/trimmed/20-A_Tajikistan_12-928_2023_trimmed_1_fastqc.html",
                "01_qc/fastqc/trimmed/20-A_Tajikistan_12-928_2023_trimmed_1_fastqc.zip",
                "01_qc/fastqc/trimmed/20-A_Tajikistan_12-928_2023_trimmed_2_fastqc.html",
                "01_qc/fastqc/trimmed/20-A_Tajikistan_12-928_2023_trimmed_2_fastqc.zip",
                "01_qc/fastqc/trimmed/51-A_Spain_1810_2024_S51_trimmed_1_fastqc.html",
                "01_qc/fastqc/trimmed/51-A_Spain_1810_2024_S51_trimmed_1_fastqc.zip",
                "01_qc/fastqc/trimmed/51-A_Spain_1810_2024_S51_trimmed_2_fastqc.html",
                "01_qc/fastqc/trimmed/51-A_Spain_1810_2024_S51_trimmed_2_fastqc.zip",
                "01_qc/trimming",
                "01_qc/trimming/2-A_SouthSudan_1029_2024.fastp.html",
                "01_qc/trimming/2-A_SouthSudan_1029_2024.fastp.json",
                "01_qc/trimming/20-A_Tajikistan_12-928_2023.fastp.html",
                "01_qc/trimming/20-A_Tajikistan_12-928_2023.fastp.json",
                "01_qc/trimming/51-A_Spain_1810_2024_S51.fastp.html",
                "01_qc/trimming/51-A_Spain_1810_2024_S51.fastp.json",
                "pipeline_info"
            ],
            [
                "viral_reference.merged.fasta:md5,887fe7e90382b03d3e9aabb2b03f2ede",
                "viral_reference.merged.fasta.fai:md5,c1207acbd8f4d0ce7d301a735c40b378",
                "2-A_SouthSudan_1029_2024_raw_1_fastqc.html:md5,8583595029269c0947c6282c88048596",
                "2-A_SouthSudan_1029_2024_raw_2_fastqc.html:md5,c60d86ba6545aa112b6adb55d6ab44e1",
                "20-A_Tajikistan_12-928_2023_raw_1_fastqc.html:md5,ec1ae907cb9950fb93ca2a948819fbad",
                "20-A_Tajikistan_12-928_2023_raw_2_fastqc.html:md5,908981e96259e59529b82aedd220cdd0",
                "51-A_Spain_1810_2024_S51_raw_1_fastqc.html:md5,7901092ff5c099f23de5e054bfd60858",
                "51-A_Spain_1810_2024_S51_raw_2_fastqc.html:md5,3d87ffc47f115d2655732c6e4b7feb15",
                "2-A_SouthSudan_1029_2024_trimmed_1_fastqc.html:md5,925237375f88cd395efd1b9bf55a3ca1",
                "2-A_SouthSudan_1029_2024_trimmed_2_fastqc.html:md5,4c342d9eeb688e0a0cb4c07ce96435ed",
                "20-A_Tajikistan_12-928_2023_trimmed_1_fastqc.html:md5,dbf829e9a1a5f3468bd82ce91e4c5c90",
                "20-A_Tajikistan_12-928_2023_trimmed_2_fastqc.html:md5,20513a70776cb928245038c4be9841c4",
                "51-A_Spain_1810_2024_S51_trimmed_1_fastqc.html:md5,8edd1878339c4c05728808537e08262c",
                "51-A_Spain_1810_2024_S51_trimmed_2_fastqc.html:md5,3fdb6381f84d70a3de78b920f45ad5c2",
                "2-A_SouthSudan_1029_2024.fastp.json:md5,279c4b44ba0d02affaa592a4b7d7445b",
                "20-A_Tajikistan_12-928_2023.fastp.json:md5,2631b7d55b7db93da3d9fd1cc77483e2",
                "51-A_Spain_1810_2024_S51.fastp.json:md5,6d1bb1a0061398e3f15e6c519f640e32"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-08T14:30:15.471308"
    },
    "preprocess_cat_fastq_singles": {
        "content": [
            4,
            [
                "00_reference",
                "00_reference/SARS-CoV-2.fasta.fai",
                "pipeline_info"
            ],
            [
                "SARS-CoV-2.fasta.fai:md5,6806a14930eab7e2de4e9776d79bcab2"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-14T15:35:47.277449"
    },
    "preprocess_trim_nanopore": {
        "content": [
            6,
            [
                "00_reference",
                "00_reference/SARS-CoV-2.fasta.fai",
                "01_qc",
                "01_qc/toulligqc",
                "01_qc/toulligqc/FAY66992_BC15_ONT",
                "01_qc/toulligqc/FAY66992_BC15_ONT/images",
                "01_qc/toulligqc/FAY66992_BC15_ONT/images/Channel_occupancy_of_the_flowcell.html",
                "01_qc/toulligqc/FAY66992_BC15_ONT/images/Correlation_between_read_length_and_PHRED_score.html",
                "01_qc/toulligqc/FAY66992_BC15_ONT/images/Distribution_of_read_lengths.html",
                "01_qc/toulligqc/FAY66992_BC15_ONT/images/PHRED_score_density_distribution.html",
                "01_qc/toulligqc/FAY66992_BC15_ONT/images/PHRED_score_distribution.html",
                "01_qc/toulligqc/FAY66992_BC15_ONT/images/PHRED_score_over_time.html",
                "01_qc/toulligqc/FAY66992_BC15_ONT/images/Read_count_histogram.html",
                "01_qc/toulligqc/FAY66992_BC15_ONT/images/Read_length_over_time.html",
                "01_qc/toulligqc/FAY66992_BC15_ONT/images/Yield_plot_through_time.html",
                "01_qc/toulligqc/FAY66992_BC15_ONT/images/plotly.min.js",
                "01_qc/toulligqc/FAY66992_BC15_ONT/report.data",
                "01_qc/toulligqc/FAY66992_BC15_ONT/report.html",
                "01_qc/toulligqc/FAY66992_BC23_ONT",
                "01_qc/toulligqc/FAY66992_BC23_ONT/images",
                "01_qc/toulligqc/FAY66992_BC23_ONT/images/Channel_occupancy_of_the_flowcell.html",
                "01_qc/toulligqc/FAY66992_BC23_ONT/images/Correlation_between_read_length_and_PHRED_score.html",
                "01_qc/toulligqc/FAY66992_BC23_ONT/images/Distribution_of_read_lengths.html",
                "01_qc/toulligqc/FAY66992_BC23_ONT/images/PHRED_score_density_distribution.html",
                "01_qc/toulligqc/FAY66992_BC23_ONT/images/PHRED_score_distribution.html",
                "01_qc/toulligqc/FAY66992_BC23_ONT/images/PHRED_score_over_time.html",
                "01_qc/toulligqc/FAY66992_BC23_ONT/images/Read_count_histogram.html",
                "01_qc/toulligqc/FAY66992_BC23_ONT/images/Read_length_over_time.html",
                "01_qc/toulligqc/FAY66992_BC23_ONT/images/Yield_plot_through_time.html",
                "01_qc/toulligqc/FAY66992_BC23_ONT/images/plotly.min.js",
                "01_qc/toulligqc/FAY66992_BC23_ONT/report.data",
                "01_qc/toulligqc/FAY66992_BC23_ONT/report.html",
                "pipeline_info"
            ],
            [
                "SARS-CoV-2.fasta.fai:md5,6806a14930eab7e2de4e9776d79bcab2"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-14T15:42:16.912363"
    }
}